### PR TITLE
Added adoption status to user

### PR DESCRIPTION
**Pull request by edwoodward**
Data comes from Salesforce and is available via the user API

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -88,6 +88,10 @@ def is_test?
     !!is_test
   end
 
+  def using_openstax?
+    false
+  end
+
```
Comment (BryanEliDimas) [inline app/models/user.rb]: Defining this is not necessary because, I believe, Rails dynamically defines it the moment you add the database column `using_openstax` to the users table. It actually might be problematic if you define it here as it will always be false regardless of the actual value in the database for `using_openstax`.

File: app/routines/update_user_salesforce_info.rb [hunk 2]
```diff
@@ -5,6 +5,13 @@ class UpdateUserSalesforceInfo
     'Career School/For-Profit (2)'
   ]
 
+  ADOPTION_STATUSES = {
+    "Current Adopter" => true,
+    "Future Adopter" => true,
+    "Past Adopter" => false,
+    "Not Adopter" => false
+  }
+
```
Comment (BryanEliDimas) [inline app/routines/update_user_salesforce_info.rb]: 🤔 How do I know that these exact strings are how they will come always come from Salesforce? If they are correct today, case-sensitive, separated by a space, what if they change in the future? I think there should be an integration test for this, at least; that would appease my concerns.

File: db/schema.rb [hunk 3]
```diff
@@ -247,10 +247,10 @@
...
     t.string "lead_application_source", default: "", null: false
     t.boolean "confidential", default: true, null: false
-    t.boolean "can_access_private_user_data", default: false, null: false
-    t.boolean "can_find_or_create_accounts", default: false, null: false
-    t.boolean "can_message_users", default: false, null: false
-    t.boolean "can_skip_oauth_screen", default: false, null: false
+    t.boolean "can_access_private_user_data", default: false
+    t.boolean "can_find_or_create_accounts", default: false
+    t.boolean "can_message_users", default: false
+    t.boolean "can_skip_oauth_screen", default: false
```
Comment (BryanEliDimas) [inline db/schema.rb]: I don't know why the `null: false` went away. I think this is not correct. You might've ran a migration at one point which modified your local database, and then switched back to your branch, ran another migration and the changes from the other branch remained? Maybe? I think you can fix this by simply reverting these lines in your schema.

File: db/schema.rb [hunk 4]
```diff
@@ -333,6 +333,7 @@
     t.citext "support_identifier", null: false
     t.boolean "is_test"
     t.integer "school_type", default: 0, null: false
+    t.boolean "using_openstax", default: false
```
Comment (BryanEliDimas) [inline db/schema.rb]: I was going to suggest that you rename this column to `is_usinng_openstax` because that is the Rails convention but... I see that `using_openstax` is already being used all throughout the codebase (I don't know why) so... I'll leave it up to you to decide.

File: app/models/user.rb (hunk 1 above)
Comment (edwoodward) [reply app/models/user.rb]: @BryanEliDimas I'm removing it, but just for my understanding, why do other boolean values in that model have the function defined, but this one should not?

Comment (BryanEliDimas) [reply app/models/user.rb]: None of the methods correspond to a database field by the same name without the `?` so we must define them, except for `is_test`, in which case I don't know the reason why 🤔. It looks like the intent was to cast it into a boolean (because of the double bang `!!`) but it's already a boolean in the database so in that case, defining that method is unnecessary but **not** problematic because it gets the value from the DB instead of a hardcoded `true` or `false`.

Comment (BryanEliDimas) [reply app/models/user.rb]: We could (and should attempt to) remove `def is_test?`, and the tests should still pass. If they don't, then I must be wrong **or** it doesn't work because of some corner case with a gem.

Comment (edwoodward) [reply app/models/user.rb]: Removed

File: app/routines/update_user_salesforce_info.rb (hunk 2 above)
Comment (edwoodward) [reply app/routines/update_user_salesforce_info.rb]: Tests added

File: db/schema.rb (hunk 3 above)
Comment (edwoodward) [reply db/schema.rb]: Fixed

File: db/schema.rb (hunk 4 above)
Comment (edwoodward) [reply db/schema.rb]: Leaving it as is.
